packages/babel-traverse/src/path/modification.ts

```diff
@@ -285,7 +285,11 @@ export function updateSiblingKeys(
   const paths = getCachedPaths(this.hub, this.parent) || ([] as never[]);
 
   for (const [, path] of paths) {
-    if (typeof path.key === "number" && path.key >= fromIndex) {
+    if (
+      typeof path.key === "number" &&
+      path.listKey === this.listKey &&
```

Comment:
Can we compare the `.container` instead? It should be faster than string comparison.

Reply:
@JLHwung my concern (and maybe this isn't valid) was that potentially during a mutation `container` got changed.
I'm happy to change it if you think that's fine?

Reply:
The `updateSiblingKeys` API is used by `path.containerInsert()` and `path.remove()`. None of them should reassign `.container`, although the elements within `container` will definitely be added or removed.

Of course one can manually changes the AST and setup the node paths again so that `.container` is changed, but so does `listKey`. In this case the NodePath methods do not guarantee they work.

Reply:
@JLHwung I went ahead and updated to check container instead. LMK if there's anything else you'd like to see!